src/main/java/uk/gov/hmcts/ccd/domain/service/globalsearch/GlobalSearchParser.java

```diff
@@ -0,0 +1,92 @@
+import uk.gov.hmcts.ccd.domain.service.common.AccessControlService;
+import uk.gov.hmcts.ccd.domain.service.common.CaseTypeService;
+import uk.gov.hmcts.ccd.domain.service.common.SecurityClassificationService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+@Component
+@Slf4j
+public class GlobalSearchParser {
+
+    private static final String FIELD_SEPARATOR = ".";
+    private final UserRepository userRepository;
+    private final CaseTypeService caseTypeService;
+    private final SecurityClassificationService securityClassificationService;
+
+    @Autowired
+    public GlobalSearchParser(@Qualifier(CachedUserRepository.QUALIFIER)
+                                  UserRepository userRepository,
+                              CaseTypeService caseTypeService,
+                              SecurityClassificationService securityClassificationService) {
+        this.userRepository = userRepository;
+        this.caseTypeService = caseTypeService;
+        this.securityClassificationService = securityClassificationService;
+    }
+
+    public List<CaseDetails> filterCases(List<CaseDetails> results, SearchCriteria request) {
+        List<String> fields = findFieldsToFilter(request);
+        results.removeIf(caseDetails -> !authorised(fields, caseDetails));
+
+        return results;
+    }
+
+    private boolean authorised(List<String> fields, CaseDetails caseDetails) {
+        CaseTypeDefinition caseTypeDefinition = caseTypeService.getCaseType(caseDetails.getCaseTypeId());
+        boolean condition = true;
+        Optional<SecurityClassification> userClassificationOpt =
+            securityClassificationService.getUserClassification(caseDetails.getJurisdiction());
+        for (String field : fields) {
+            Optional<CaseFieldDefinition> caseFieldDefinition =
+                (field.contains(FIELD_SEPARATOR)) ? caseTypeDefinition.getComplexSubfieldDefinitionByPath(field)
+                    : caseTypeDefinition.getCaseField(field);
+            if (caseFieldDefinition.isPresent() && (!AccessControlService
+                .hasAccessControlList(userRepository.getUserRoles(), CAN_READ,
+                    caseFieldDefinition.get().getAccessControlLists()))){
+                condition = false;
+                break;
+            }
+            else if (!securityClassificationService
+                .userHasEnoughSecurityClassificationForField(caseTypeDefinition.getJurisdictionId(),
+                    caseTypeDefinition, caseFieldDefinition.get().getId())) {
+                condition = false;
+                break;
+            }
+
+        }
+        return condition;
+    }
+
+    private List<String> findFieldsToFilter(SearchCriteria request) {
+        List<String> fields = new ArrayList<>();
+        if (request.getCaseManagementBaseLocationIds() != null) {
```

Comment:
```suggestion
        if (!CollectionUtils.isEmpty(request.getCaseManagementBaseLocationIds())) {
```
Think it's worth adding a check for not empty too (this does both)